### PR TITLE
Adaptive cards

### PR DESCRIPTION
**Pull request by kwinto**
This PR adds support for Adaptive Cards.

Also:
- Improves demo page
- Moves demo page to test to keep the code only relevant for the package in src

To test:

`npm run dev`

`press 'o' + <Enter>`

Note: there seem to be an issue related to Swiper+Adaptive Cards dependencies, it currently breaks test runs.

### CODE REVIEW

**Comment by fabclj** [comment]
I like the new demo page, but it would be great to integrate all messages types.
Also it would be great to solve the issue about swiper and adaptive card.

**Comment by fabclj** [comment]
I have run the adaptive card on webchat locally. I list here the first findings:

Screenshot from 2024-01-11 12-11-03


- Primary button should have height 44px and cursor: pointer
- The card width should be max 295px like all messages
- Textarea scrollbar is not styled as Figma requirments

File: src/messages/AdaptiveCards/index.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			return _defaultPreview;
+		}
+		return _plugin || _webchat;
+	};
+
+	const cardPayload = getCardPayload(message);
+
+	const onExecuteAction = useCallback(
+		(action: Action) => {
+			console.log("onExecuteAction", action);
```
Comment (fabclj) [inline src/messages/AdaptiveCards/index.tsx:73]: please remove debug rest

File: src/messages/AdaptiveCards/index.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			},
+		},
+	},
+};
+
+export const AdaptiveCards = () => {
+	const { message, action: onSendMessage, config } = useMessageContext();
+
+	const getCardPayload = (message: IMessage) => {
+		//@ts-ignore
```
Comment (fabclj) [inline src/messages/AdaptiveCards/index.tsx:52]: we should refine the TS declarations at some point.

File: package.json [hunk 3]
```diff
@@ -25,35 +25,43 @@
...
 		"dompurify": "^3.0.6",
 		"react-player": "^2.13.0",
+		"remarkable": "^2.0.1",
 		"swiper": "^11.0.4"
 	},
 	"peerDependencies": {
 		"react": "^17.x || 18.x",
 		"react-dom": "^17.x || 18.x"
+	},
+	"overrides": {
```
Comment (fabclj) [inline package.json:62]: this overrides should be also defined on the consumer apps, for example in webchat v3

Comment (kwinto) [reply package.json:62]: We will address this in later PRs.

File: src/messages/AdaptiveCards/index.tsx (hunk 2 above)
Comment (kwinto) [reply src/messages/AdaptiveCards/index.tsx:52]: This is not an intentional error. Like the one which is explicitly false, which you would use in e.g. test case :)

We should do a next round of interfaces improvements at some point.

File: src/messages/AdaptiveCards/index.tsx (hunk 1 above)
Comment (kwinto) [reply src/messages/AdaptiveCards/index.tsx:73]: fixed

**Comment by kwinto** [comment]
> * Primary button should have height 44px and cursor: pointer
Fixed

> * The card width should be max 295px like all messages
Full width for AC messages is specified in Figma

> * Textarea scrollbar is not styled as Figma requirments
I fixed it as far as it is possible with CSS. We lack the possibility to have a margin and coloured "track" at the same time on a scrollbar. 

Further improvements here will need a custom AC render or some kind of field override, as we are at the limit of what is possible with to do with pure CSS.

**Comment by kwinto** [comment]
> I like the new demo page, but it would be great to integrate all messages types. Also it would be great to solve the issue about swiper and adaptive card.

Tracked in new sotries for the next sprint.

File: src/messages/AdaptiveCards/styles.module.css [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	max-width: 100%;
+	padding: 16px !important;
+}
+
+.adaptivecard-wrapper > *:focus {
+	outline: 1px solid var(--primary-color);
+	border-color: var(--primary-color);
+}
+
+.adaptivecard-wrapper button {
```
Comment (fabclj) [inline src/messages/AdaptiveCards/styles.module.css:44]: I think we are also missing the button styles when is disabled

**Comment by fabclj** [comment]
Text area now has scrollbar but is still not fitting the design requirements

Screenshot from 2024-01-11 16-20-25

File: src/messages/AdaptiveCards/styles.module.css (hunk 4 above)
Comment (kwinto) [reply src/messages/AdaptiveCards/styles.module.css:44]: fixed

**Review by fabclj — APPROVED** [review]

**Pull request merged**